### ISSUE TITLE
EmptyLineSeparator check does not validate newlines before comments in Interfaces

### ISSUE THREAD

**Issue by mhagnumdw** [checkstyle/checkstyle#16564]
This works on classes, great. But it doesn't work on interfaces?

I have read check documentation: https://checkstyle.org/checks/whitespace/emptylineseparator.html#EmptyLineSeparator
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
$ java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)

$ javac -version
javac 17.0.2

$ javac MyInterface.java
# no output, exit code 0

$ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="charset" value="UTF-8"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="allowNoEmptyLineBetweenFields" value="true"/>
      <property name="allowMultipleEmptyLines" value="false"/>
      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
    </module>
  </module>
</module>

$ cat MyInterface.java
package io.mhagnumdw;

public interface MyInterface {


    /**
     * Test.
     */
    void myMethod();

}

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"

$ java $RUN_LOCALE -jar checkstyle-10.21.4-all.jar -c config.xml MyInterface.java
Starting audit...
Audit done.
```

---

It should be reported that there is more than one consecutive blank line. Lines 4 and 5. There should be a WARN from EmptyLineSeparator. The same case, being a class, works correctly.

related issue: https://github.com/checkstyle/checkstyle/issues/5981

**Comment by AyushInKC** [comment]
Instead of multiline EmptyLineSeparator we can use RegexpMultiline  This will flag any occurrence of three or more consecutive newlines, including in interfaces. 

The RegexpMultiline  check detects multiple consecutive blank lines (3 or more) anywhere in the code, including interfaces, classes, and methods. Since EmptyLineSeparator does not always enforce blank line rules inside interfaces (especially with Javadoc), this acts as a fallback to ensure consistency.

**Comment by romani** [comment]
please report issue as destibed in https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

we need CLI output  and full listing of config and java file.

**Comment by mhagnumdw** [comment]
@romani , done. Let me know if you need anything else.

**Comment by romani** [comment]
@SteLeo1602 , can you help us to confirm that it work with WARN on classes ?
and please help to fix it.

**Comment by SteLeo1602** [comment]
On it.

**Comment by SteLeo1602** [comment]
> [@SteLeo1602](https://github.com/SteLeo1602) , can you help us to confirm that it work with WARN on classes ? and please help to fix it.

```bash
PS D:\Projects\work_checkstyle> cat config.xml
<?xml version="1.0" encoding="UTF-8"?>                                                                                           
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">                                                                                                          
  <property name="severity" value="warning"/>                                                                                    
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="allowNoEmptyLineBetweenFields" value="true"/>
      <property name="allowMultipleEmptyLines" value="false"/>
      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
    </module>
  </module>
</module>
PS D:\Projects\work_checkstyle> cat example.java
package io.mhagnumdw;
public class MyClass {


    /**
     * Test.
     */
    void myMethod();

}
PS D:\Projects\work_checkstyle> java -jar checkstyle-10.21.4-all.jar -c config.xml example.java 
Starting audit...
Audit done.
PS D:\Projects\work_checkstyle> 
```

Works on classes.

**Comment by SteLeo1602** [comment]
> > [@SteLeo1602](https://github.com/SteLeo1602) , can you help us to confirm that it work with WARN on classes ? and please help to fix it.
> 
> PS D:\Projects\work_checkstyle> cat config.xml
> <?xml version="1.0" encoding="UTF-8"?>                                                                                           
> <!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
> <module name="Checker">                                                                                                          
>   <property name="severity" value="warning"/>                                                                                    
>   <property name="fileExtensions" value="java, properties, xml"/>
>   <module name="TreeWalker">
>     <module name="EmptyLineSeparator">
>       <property name="allowNoEmptyLineBetweenFields" value="true"/>
>       <property name="allowMultipleEmptyLines" value="false"/>
>       <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
>     </module>
>   </module>
> </module>
> PS D:\Projects\work_checkstyle> cat example.java
> package io.mhagnumdw;
> public class MyClass {
> 
> 
>     /**
>      * Test.
>      */
>     void myMethod();
> 
> }
> PS D:\Projects\work_checkstyle> java -jar checkstyle-10.21.4-all.jar -c config.xml example.java 
> Starting audit...
> Audit done.
> PS D:\Projects\work_checkstyle> 
> No errors were reported on classes as well.

@mhagnumdw Could you please explain why do you think it works on classes as it doesn't work in the case of my comment code either

**Comment by SteLeo1602** [comment]
```bash
PS D:\Projects\work_checkstyle> cat Config.xml  
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <property name="severity" value="warning"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="EmptyLineSeparator">
      <property name="allowNoEmptyLineBetweenFields" value="true"/>
      <property name="allowMultipleEmptyLines" value="false"/>
      <property name="allowMultipleEmptyLinesInsideClassMembers" value="false"/>
    </module>
  </module>
PS D:\Projects\work_checkstyle> cat Example.java  
package io.mhagnumdw;

public interface Example {


    void myMethod();

}
PS D:\Projects\work_checkstyle> java -jar checkstyle-10.21.4-all.jar -c config.xml example.java
Starting audit...
[WARN] D:\Projects\work_checkstyle\example.java:6:5: 'METHOD_DEF' has more than 1 empty lines before. [EmptyLineSeparator]
Audit done.
```

Correctly report violations without Javadoc present in the code. The actual defect is in Javadoc presence and not between Interface and Classes

**Comment by mhagnumdw** [comment]
> [@mhagnumdw](https://github.com/mhagnumdw) Could you please explain why do you think it works on classes as it doesn't work in the case of my comment code either

@SteLeo1602 , You are right, in classes it also presents the same problem. Somehow I was mistaken.

P.S.: your class in https://github.com/checkstyle/checkstyle/issues/16564#issuecomment-2745399644 is invalid because the method body is missing

**Comment by nrmancuso** [comment]
Closed via #16645, thanks @SteLeo1602 !
